File: javascript/bob/bob.js
let Bob = function() {}

Bob.prototype.hey = function(input) {
  if (input.toUpperCase() === input && input.toLowerCase() !== input) {
    return "Whoa, chill out!"
  } else if (input.endsWith("?")) {
    return "Sure."
  } else if (input.trim()) {
    return "Whatever."
  }
  return "Fine. Be that way!"
}

module.exports = Bob

File: scala/bob/src/main/scala/bob.scala
class Bob {
  def hey(phrase: String): String = phrase match {
    case s if isBlank(s) => "Fine. Be that way!"
    case s if isShout(s) => "Whoa, chill out!"
    case s if isQuestion(s) => "Sure."
    case _ => "Whatever."
  }

  private def isBlank(phrase: String): Boolean = phrase.trim().isEmpty()
  private def isShout(phrase: String): Boolean = phrase.toUpperCase() == phrase && phrase.toLowerCase() != phrase
  private def isQuestion(phrase: String): Boolean = phrase.endsWith("?")
}
